pkg/domain/infra/abi/play.go

```diff
@@ -783,6 +788,21 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 		containers = append(containers, ctr)
 	}
 
+	// Disconnect the infra container network and re-connect it with the original container names from the kube yaml
+	// as aliases for it. This will allow network to work with both just containerName as well as containerName-podName
+	// In the future, we want to extend this to the CLI as well, where the name of the container created will not have
+	// the podName appended to it, but this is a breaking change and will be done in podman 5.0
+	infraCtr, err := pod.InfraContainer()
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := infraCtr.NetworkDisconnect(infraCtr.Name(), kubeDefaultNetwork, true); err != nil {
+		return nil, nil, err
+	}
+	if err = infraCtr.NetworkConnect(infraCtr.Name(), kubeDefaultNetwork, nettypes.PerNetworkOptions{Aliases: ctrNameAliases}); err != nil {
+		return nil, nil, err
+	}
+
```

Comment:
Why the extra disconnect/connect? This will just create unnecessary db update overhead and as we found out this is slow.

You should be able to set the aliases on pod creation.

Also this will fail if the network mode is not set to bridge.

Reply:
Fixed it to happen during pod creation instead.